src/PowerPing/DisplayUpdateLimiter.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+MIT License - PowerPing 
+
+Copyright (c) 2018 Matthew Carney
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+using System;
+using System.Diagnostics;
+
+namespace PowerPing
+{
+    public class DisplayUpdateLimiter
+    {
+        private readonly long minimumUpdateIntervalTicks;
+        private long? lastUpdateTimestamp;
+
+        public DisplayUpdateLimiter(TimeSpan minimumUpdateInterval)
+        {
+            minimumUpdateIntervalTicks = Helper.TimeSpanToStopwatchTicks(minimumUpdateInterval.Ticks);
+        }
+
+        public TimeSpan? ElapsedSinceLastUpdate { get; private set; }
+
+        public bool RequestUpdate()
+        {
+            long currentTimestamp = Stopwatch.GetTimestamp();
+            long? elapsed = currentTimestamp - lastUpdateTimestamp;
```

Comment:
Question mark after the type indicates that it's a nullable type. Of course classes (e.g. strings) are inherently nullable, but anything that's a value type (struct) needs the question mark if you want to assign it null.

Maybe this code is being too 'clever' though. I am relying on the fact that math like "a - b" may result in null if either 'a' or 'b' is null. Because lastUpdateTimestamp may be null if this is the first update.